haskell-ci.cabal

```diff
@@ -137,7 +137,7 @@ library haskell-ci-internal
   build-depends:
     , base          >=4.10     && <4.15
     , bytestring    ^>=0.10.8.1
-    , Cabal         ^>=3.0
+    , Cabal         ^>=3.0 || ^>= 3.2
```

Comment:
Just drop `3.0` support. It's easier for us.

Reply:
I can do that. I have one question, however. Currently, `ParseError` is defined like this:

https://github.com/haskell-CI/haskell-ci/blob/7d6bf41f3338750b8f4b069cbd4889fe509c1e73/cabal-install-parsers/src/Cabal/Parse.hs#L50-L55

If we drop `Cabal-3.0` support, you might think that the simplest solution is to change the type of `peErrors` from `[PError]` to `NonEmpty PError`, since that would match the corresponding change in `runParseResult`'s type signature. However, that won't quite work, since there is a line of code in `haskell-ci` that specifically uses an empty list for `peErrors`:

https://github.com/haskell-CI/haskell-ci/blob/1e5d6138e51bfa90f03a33480ac8c71453153aa8/src/HaskellCI/ParsecUtils.hs#L36-L39

What should be done about this line of code?

Reply:
Also, should we drop `Cabal-3.0` support for both `cabal-install-parsers` _and_ `haskell-ci`, or just `haskell-ci`?

Reply:
I'd say both. As far as they are in the same repository we want be able to test that older Cabal support doesn't rot.

As for `peErrors`, I'd try to do 

```
data ParseError f = ParseError
    ...
   , peErrors = f C.PError
```

and make `renderParseError` require `Foldable f`.

If you mind trying that out, it would be great. Otherwise you can leave it as list.

Reply:
I've required `Cabal-3.2` everywhere, and `ParseError` is now parameterized by `f`, the list type.